x/metoken/README.md

```diff
@@ -0,0 +1,326 @@
+- IST  deviation from targeted allocation in percentages: (0.33333 - 0.60483)/(0.33333 / 100) = -0.2715 / 0.0033333 = 81.45081%
+
+- USDT swap fee: 0.2 + [27.42027*(0.2/100)]  = 0.25484
+- USDC swap fee: 0.2 + [54.03354*(0.2/100)]  = 0.30806
+- IST  swap fee: 0.2 + [-81.45081*(0.2/100)] = 0.03709
+```
+
+Another example with an edge case where the min and max fee are used:
+``` makefile
+- Fee:
+  - Min: 0.01
+  - Balanced: 0.3
+  - Max: 0.8
+
+- USDT:
+  - target_allocation: 0.25
+- USDC: 
+  - target_allocation: 0.25
+- IST: 
+  - target_allocation: 0.25
+- MSK: 
+  - target_allocation: 0.25
+
+After several swaps the index has 3500 USDT, 100 USDC, 300 IST and 0 MKS. Total supply: 3900. 
+
+Calculations for swap:
+- USDT deviation from targeted allocation in percentages: (0.89743 - 0.25)/(0.25 / 100) =  0.64743  / 0.0025 =  258.972%
+- USDC deviation from targeted allocation in percentages: (0.02564 - 0.25)/(0.25 / 100) = -0.22436  / 0.0025 = -89.744%
+- IST  deviation from targeted allocation in percentages: (0.07692 - 0.25)/(0.25 / 100) = -0.17308  / 0.0025 = -69.232%
+- MKS  deviation from targeted allocation in percentages: (0.0 - 0.25)/(0.25 / 100)     = -0.25     / 0.0025 = -100%
+
+- USDT swap fee: 0.3 + [258.972*(0.3/100)]   = 1.07691 -> This exceedes the max fee (0.8). In this case the max fee will be used.
+- USDC swap fee: 0.3 + [-89.744*(0.3/100)]   = 0.03076
+- IST  swap fee: 0.3 + [-69.232*(0.3/100)]   = 0.09230
+- MKS  swap fee: 0.3 + [-100*(0.3/100)]      = 0       -> This is below the min fee (0.01). For this swap the min fee will be applied.
+
+Calculations for redemption:
+- USDT deviation from targeted allocation in percentages: (0.25 - 0.89743)/(0.25 / 100) = -0.64743  / 0.0025 = -258.972%
+- USDC deviation from targeted allocation in percentages: (0.25 - 0.02564)/(0.25 / 100) =  0.22436  / 0.0025 =  89.74358%
+- IST  deviation from targeted allocation in percentages: (0.25 - 0.07692)/(0.25 / 100) =  0.17308  / 0.0025 =  69.232%
+- MKS  deviation from targeted allocation in percentages: (0.25 - 0.0)/(0.25 / 100)     =  0.25     / 0.0025 =  100%
+
+- USDT swap fee: 0.3 + [-258.972*(0.3/100)] = -0.47691 -> This is below the min fee (0.01) and also the fee can't be negative. For this swap the min fee will be applied.
+- USDC swap fee: 0.3 + [89.744*(0.3/100)]   = 0.56923
+- IST  swap fee: 0.3 + [69.232*(0.3/100)]   = 0.50769
+- MKS  swap fee: 0.3 + [100*(0.3/100)]      = 0.6      -> In this case the redeption is not possible since there is no MKS liquidity anyway.
+```
+
+### Reserves
+
+The `metoken` module will have its own reserves to stabilize the processing of the withdrawals. A portion of
+every swap will be transferred to the reserves and a percentage of every withdrawal will be taken from the reserves.
+This portion is determined by the parameters of every asset. Every charged fee will be also transferred to the
```

Comment:
All fees shouldn't be added to the reserve, we want them to eventually be shared with Umee stakers. So for now we need to have a separate fee address.